### DISCUSSION TITLE
Documentation for Cloudflare misleading

### DISCUSSION THREAD

**Discussion by devmaxde** [go-acme/lego#2327] [Q&A] [answered]
### Welcome

- [X] Yes, I've searched similar issues on GitHub and didn't find any.

### How do you use lego?

Binary

### Detailed Description

Hey,

when I tried setting up the lego binary to generate a ssl cert for my domain I tried using the example from the Documentation.

The main problem here ist, that depending on the API Key Type you need to set a different environment variable. I tried the CLOUDFLARE_API_KEY which failed with a similar issue like #930.

It wasn't able to find the Zone. After changing the environment to CF_DNS_API_TOKEN it worked. When providing simple examples please make sure it's not misleading. 
(wasted 2h for this :( )

<img width="1353" alt="Bildschirmfoto 2024-11-07 um 17 29 26" src="https://github.com/user-attachments/assets/06faa7ec-dab5-4263-aedf-18b1fe4d0be5">

**Comment by ldez** [comment, marked as answer]
Hello,

as stated inside the documentation there 2 ways to use the provider:
1. `CLOUDFLARE_EMAIL` **+** `CLOUDFLARE_API_KEY`
2. `CLOUDFLARE_DNS_API_TOKEN`

With the first, you must provide `CLOUDFLARE_EMAIL` and `CLOUDFLARE_API_KEY`.

With the second, only `CLOUDFLARE_DNS_API_TOKEN` is required.

The prefix of the env vars can be `CLOUDFLARE_` or `CF_`, the 2 prefixes are handled in the same way.

For the details, the env vars prefix `CF_` is just aliasas on the `CLOUDFLARE_` prefix.

**Reply by ldez** [reply]
Also, each approach has its own requirements, which are explained in the doc:
- https://go-acme.github.io/lego/dns/cloudflare/index.html#api-keys
- https://go-acme.github.io/lego/dns/cloudflare/index.html#api-tokens

**Reply by devmaxde** [reply]
Yeah. I know that. But in the Screenshot there is no indication of that. And there will always be an idiot like me who will only try the first example. Adding another hint there won't hurt.

**Reply by ldez** [reply]
I'm sorry but we will not duplicate information because a few users over 10 years don't take the time to read the documentation.

The documentation contains all the information: there are sections for each case with an explanation.

But we will remove the example based on Cloudflare.
